bundles/org.openhab.binding.solarforecast/src/main/java/org/openhab/binding/solarforecast/internal/solcast/handler/SolcastBridgeHandler.java

```diff
@@ -0,0 +1,253 @@
+        super(bridge);
+        timeZone = tzp.getTimeZone();
+    }
+
+    @Override
+    public Collection<Class<? extends ThingHandlerService>> getServices() {
+        return List.of(SolarForecastActions.class);
+    }
+
+    @Override
+    public void initialize() {
+        SolcastBridgeConfiguration configuration = getConfigAs(SolcastBridgeConfiguration.class);
+        if (!configuration.apiKey.isBlank()) {
+            if (!configuration.timeZone.isBlank()) {
+                try {
+                    timeZone = ZoneId.of(configuration.timeZone);
+                } catch (DateTimeException e) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                            "@text/solarforecast.site.status.timezone" + " [\"" + configuration.timeZone + "\"]");
+                    return;
+                }
+            }
+            updateStatus(ThingStatus.UNKNOWN);
+            refreshJob = Optional
+                    .of(scheduler.scheduleWithFixedDelay(this::getData, 0, REFRESH_ACTUAL_INTERVAL, TimeUnit.MINUTES));
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "@text/solarforecast.site.status.api-key-missing");
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            String channel = channelUID.getIdWithoutGroup();
+            switch (channel) {
+                case CHANNEL_ENERGY_ACTUAL:
+                case CHANNEL_ENERGY_REMAIN:
+                case CHANNEL_ENERGY_TODAY:
+                case CHANNEL_POWER_ACTUAL:
+                    getData();
+                    break;
+                case CHANNEL_POWER_ESTIMATE:
+                case CHANNEL_ENERGY_ESTIMATE:
+                    forecastUpdate();
+                    break;
+            }
+        }
+    }
+
+    @Override
+    public void dispose() {
+        refreshJob.ifPresent(job -> job.cancel(true));
+    }
+
+    /**
+     * Get data for all planes. Protect plane list from being modified during update
+     */
+    public synchronized void getData() {
+        if (planes.isEmpty()) {
+            logger.debug("No PV plane defined yet");
+            return;
+        }
+        ZonedDateTime now = ZonedDateTime.now(getTimeZone());
+        List<QueryMode> modes = List.of(QueryMode.Average, QueryMode.Pessimistic, QueryMode.Optimistic);
+        modes.forEach(mode -> {
+            String group = GROUP_AVERAGE;
+            switch (mode) {
+                case Average:
+                    group = GROUP_AVERAGE;
+                    break;
+                case Optimistic:
+                    group = GROUP_OPTIMISTIC;
+                    break;
+                case Pessimistic:
+                    group = GROUP_PESSIMISTIC;
+                    break;
+            }
```

Comment:
Can be shortened (if you wish):
```suggestion
            String group = switch (mode) {
                case Average -> GROUP_AVERAGE;
                case Optimistic -> GROUP_OPTIMISTIC;
                case Pessimistic -> GROUP_PESSIMISTIC;
            };
```

Reply:
Done